src/XAD/TypeTraits.hpp

```diff
@@ -0,0 +1,45 @@
+/*******************************************************************************
```

Comment:
you need to add `#pragma once` at the top of all new headers you added. To avoid multiple inclusion.